src/liblsquic/lsquic_handshake.c

```diff
@@ -1143,15 +1143,15 @@ static int parse_hs_data (struct lsquic_enc_session *enc_session, uint32_t tag,
         break;
 
     case QTAG_SCID:
-        if (len != SCID_LENGTH)
+        if (len < SCID_LENGTH)
```

Comment:
I don't see how this would work with SCID lengths larger than 16, as `SCID_LENGTH` is 16.  Did you forget to commit a file?

Reply:
Actually, I take it back -- I see what has changed.  If `len` larger than `SCID_LENGTH` (in your case, 32 bytes), the memory copy will overwrite the next buffer in the struct.

Reply:
> Actually, I take it back -- I see what has changed. If `len` larger than `SCID_LENGTH` (in your case, 32 bytes), the memory copy will overwrite the next buffer in the struct.

What we hope lsquic to compatible scene: 

LSQUIC send REJ but not PRST when client send CHLO with 32B scid, So I only change the logic of "cas QTAG_SCID".

Reply:
> I don't see how this would work with SCID lengths larger than 16, as `SCID_LENGTH` is 16. Did you forget to commit a file?

We don't use 32B scid any more after we use lsquic, but we have lots of client which have store old scid in their cache. 

Before SCFG expired, client will use those 32B scid, and if we update to lsquic, lsquic will send PRST but not REJ and then connect failed.

Reply:
I misread the code twice!  I now understand what it is you're trying to achieve, thank you for the explanation.

Reply:
> I misread the code twice! I now understand what it is you're trying to achieve, thank you for the explanation.

This is my fault, I didn't describe the purpose clearly, sorry!